### PR TITLE
Standardized hashing

### PR DESCRIPTION
**Pull request by Alexbay218**
Adding a proposed standard on hashing boards and full states. This is designed for use in a future standardized puzzle format.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -422,6 +424,64 @@ w. (0T1)d2(0T1)d4
...
+For purposes of sorting, timeline -0 is considered lower than timeline +0.
+
+For strictness, extracting 5DFEN strings should standardize on move sensitivity. The following pieces should be marked as unmoved if needed (the rest of the pieces should not include the unmoved marking):
+
+  - Pawn
+  - Rook
+  - King
+  - Brawn
+  - Common King
+  - Royal Queen
```
Comment (adri326) [inline README.md]: The common king and royal queen are not move-sensitive, so they don't need to be in this list

File: README.md [hunk 2]
```diff
@@ -422,6 +424,64 @@ w. (0T1)d2(0T1)d4
...
+  3. Extract the resulting hash as hexadecimal digits
+  4. Convert to UTF-8 string (alphabet characters should be lowercase)
+
+For full state hashing:
+
+  1. Grab the 5DFEN string of all boards, sorting boards by timeline (ascending) and then by turn (ascending)
+  2. Concatenate all of these 5DFEN strings together (keeping the order), stripping any and all whitespaces (including newlines)
+  3. Feed the resulting string (encoded as UTF-8) into the MD5 message-digest algorithm
+  4. Extract the resulting hash as hexadecimal digits
+  5. Convert to UTF-8 string (alphabet characters should be lowercase)
```
Comment (adri326) [inline README.md:446]: *Note:* this is equivalent to keeping the hash state and hashing each board in the specified order

File: README.md [hunk 3]
```diff
@@ -54,6 +53,9 @@ This repository includes a converter and previewer. For information on how to ru
   - [Export or minimal notation](#export-or-minimal-notation)
     - [Export moves](#export-moves)
     - [Export turns](#export-turns)
+  - [Hashing](#hashing)
+    - [5DFEN Stricness](#5dfen-strictness)
```
Comment (adri326) [inline README.md]: Mind the small typo (Stric**t**ness)

**Review by adri326 — COMMENTED** [review]
Looks good, I'll ask the others on their thoughts

File: README.md (hunk 2 above)
Comment (Alexbay218) [reply README.md:446]: Not sure what you mean by this

Comment (adri326) [reply README.md:446]: It's just another way to compute the hash, that only requires O(1) memory

**Pull request merged**